File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/item/RelicSongItem.java
package generations.gg.generations.core.generationscore.common.world.item;

import net.minecraft.network.chat.Component;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.TooltipFlag;
import net.minecraft.world.level.Level;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class RelicSongItem extends Item implements LangTooltip {
    private final boolean notInert;

    public RelicSongItem(Item.Properties properties, boolean notInert) {
        super(properties.jukeboxPlayable(RecordSongs.INSTANCE.getMELOETTAS_RELIC_SONG()));
        this.notInert = notInert;
    }

    @Override
    public boolean isFoil(@NotNull ItemStack stack) {
        return !notInert;
    }

    @Override
    public void appendHoverText(ItemStack stack, @Nullable Level level, List<Component> tooltipComponents, TooltipFlag isAdvanced) {
        LangTooltip.super.appendHoverText(stack, level, tooltipComponents, isAdvanced);
    }
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/world/item/RelicSongItem.kt
package generations.gg.generations.core.generationscore.common.world.item

import generations.gg.generations.core.generationscore.common.world.item.RecordSongs.MELOETTAS_RELIC_SONG
import net.minecraft.network.chat.Component
import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.TooltipFlag

class RelicSongItem(properties: Properties, private val notInert: Boolean) : Item(
    properties.jukeboxPlayable(
        MELOETTAS_RELIC_SONG
    )
),
    LangTooltip {
    override fun isFoil(stack: ItemStack): Boolean {
        return !notInert
    }

    override fun appendHoverText(
        stack: ItemStack,
        level: TooltipContext, tooltipComponents: MutableList<Component>,
        isAdvanced: TooltipFlag
    ) {
        addText(stack, level, tooltipComponents, isAdvanced)
    }
}
